core/src/com/unciv/ui/utils/CameraStageBaseScreen.kt

```diff
@@ -24,36 +24,77 @@ import java.util.HashMap
 import kotlin.concurrent.thread
 import kotlin.random.Random
 
+// Gdx Input.Keys.F1 _is_ of type Int
+data class KeyCharAndCode(val char: Char, val code: Int) {
+    // express keys with a Char value
+    constructor(char: Char): this(char.toLowerCase(), 0)
+    // express keys that only have a keyCode like F1
+    constructor(code: Int): this(Char.MIN_VALUE, code)
+    // helper for use in InputListener keyTyped()
+    constructor(event: InputEvent?, character: Char)
+            : this (
+                character.toLowerCase(),
+                if (character == Char.MIN_VALUE && event!=null) event.keyCode else 0
+            )
+}
+
+class KeyPressDispatcher {
```

Comment:
We can make `KeyPressDispatcher : HashMap<KeyCharAndCode, (() -> Unit)>`
And then we can save the get/set of the KeyCharAndCode

Reply:
I know but I preferred this variant when I considered inheritance - maybe I thought something like 'you can explicitly see everything that's going on'. You're probably right. Let's see whether git breaks this as badly as it broke the translation file...